src/Blazor.FlexGrid/Components/Configuration/MetaData/GridAnotations.cs

```diff
@@ -22,11 +22,14 @@ public bool IsMasterTable
             }
         }
 
-        public bool OnlyShowExplicitDetailTables
+        
+        public MasterDetailOptions MasterDetailOptions
         {
             get
             {
-                return entityTypeMetadata.OnlyShowExplicitDetailTables;
+                var masterDetailOptions = annotations[GridViewAnnotationNames.MasterDetailOptions];
```

Comment:
This can be `NullAnotationValue` and the casting will fail. Add check for `NullAnotationValue` and return `NullMasterDetaiOptions` object